### PR TITLE
Bug 1862700: Fix ovnkube-node aggressive memory allocation to prevent memory bloating

### PR DESCRIPTION
**Pull request by vishnoianil**
…memory bloating

**- What this PR does and why is it needed**
During the pod creation, ovnkube-node dump all the flows from br-int and grep
for the ip address of the pod to verify that pod related flows are programmed.
This can cause aggrasive and big memory allocation to store the string output
from ovs-ofctl command.
This patch optimized the ovs-ofctl dump-flow query to dump flows only related
to the pods. This will avoid dumping the massive string output and prevent
from memory bloating.

Signed-off-by: Anil Vishnoi <avishnoi@redhat.com>

### CODE REVIEW

**Comment by vishnoianil** [comment]
@dcbw PTAL

**Comment by danwinship** [comment]
/hold
don't merge while discussion is still happening on #226

**Comment by danwinship** [comment]
/retitle Bug 1862700: Fix ovnkube-node aggressive memory allocation to prevent memory bloating

File: go-controller/pkg/cni/helper_linux.go [hunk 1]
```diff
@@ -332,14 +332,7 @@ func (pr *PodRequest) ConfigureInterface(namespace string, podName string, ifInf
...
 
-	err = wait.PollImmediate(100*time.Millisecond, 20*time.Second, func() (bool, error) {
-		stdout, err := ofctlExec("dump-flows", "br-int")
-		if err != nil {
-			return false, nil
-		}
-		return strings.Contains(stdout, ifInfo.IPs[0].IP.String()), nil
-	})
-	if err != nil {
+	if err = waitForPodFlows(ifInfo.MAC.String()); err != nil {
```
Comment (abhat) [inline go-controller/pkg/cni/helper_linux.go:335]: waitForPodFlows needs to be renamed. I think a lot of comments [here](https://github.com/ovn-org/ovn-kubernetes/pull/1568/files) apply to this PR. Also, any reason why we aren't fixing this upstream first?

Comment (vishnoianil) [reply go-controller/pkg/cni/helper_linux.go:335]: @abhat why do we need to change the name? waitForPodFlows is literally waiting for pod flows to appear in the bridge.

Comment (vishnoianil) [reply go-controller/pkg/cni/helper_linux.go:335]: @abhat i think the idea was that upstream fix might take time, and we want to get this in quickly to resolve the memory bloat issue.

**Comment by vishnoianil** [comment]
@dcbw all tests are passing on this patch.

File: go-controller/pkg/cni/helper_linux.go (hunk 1 above)
Comment (dcbw) [reply go-controller/pkg/cni/helper_linux.go:335]: Yes.

**Comment by dcbw** [comment]
AWS failures seem like a whole lot of apiservers going down at 22:04:30-ish. That causes a lot of:

testcase:
```
Aug  5 22:17:55.955: FAIL: rpc error: code = Unavailable desc = transport is closing
```

apiserver:
```
I0805 22:15:15.573965      18 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I0805 22:15:15.574211      18 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I0805 22:15:15.574305      18 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I0805 22:15:15.576440      18 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
```

on the part of the apiserver, which is apparently a gRPC server issue, but it's very unclear what the apiserver is suppose

**Comment by trozet** [comment]
/hold

@vishnoianil looks like we accidentally merged a fixup in master. Can you please squash your fix up for the backports?

**Comment by vishnoianil** [comment]
> /hold
> 
> @vishnoianil looks like we accidentally merged a fixup in master. Can you please squash your fix up for the backports?

done. @dcbw can you please lgtm and approve the pr.

**Comment by ashcrow** [comment]
Note for the upcoming patch manager: This was meant to be approved in the last cycle but failed to get the label. PTAL.

**Pull request merged**